pkg/scheduler/apis/config/v1beta2/default_plugins.go

```diff
@@ -148,19 +148,16 @@ func mergePlugins(defaultPlugins, customPlugins *v1beta2.Plugins) *v1beta2.Plugi
 	return defaultPlugins
 }
 
-type pluginIndex struct {
-	index  int
-	plugin v1beta2.Plugin
-}
-
 func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2.PluginSet {
 	disabledPlugins := sets.NewString()
-	enabledCustomPlugins := make(map[string]pluginIndex)
+	// replacedCustomPlugins is a set of plugins, each of them is default plugin as well.
+	replacedCustomPlugins := sets.NewString()
```

Comment:
We can eliminate this by simply remove the entry from enabledCustomPlugins:

```diff
diff --git a/pkg/scheduler/apis/config/v1beta2/default_plugins.go b/pkg/scheduler/apis/config/v1beta2/default_plugins.go
index dd5cbb3ff10..dcdc79bb2d2 100644
--- a/pkg/scheduler/apis/config/v1beta2/default_plugins.go
+++ b/pkg/scheduler/apis/config/v1beta2/default_plugins.go
@@ -150,8 +150,6 @@ func mergePlugins(defaultPlugins, customPlugins *v1beta2.Plugins) *v1beta2.Plugi
 
 func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2.PluginSet {
 	disabledPlugins := sets.NewString()
-	// replacedCustomPlugins is a set of plugins, each of them is default plugin as well.
-	replacedCustomPlugins := sets.NewString()
 	enabledCustomPlugins := make(map[string]v1beta2.Plugin)
 	for _, disabledPlugin := range customPluginSet.Disabled {
 		disabledPlugins.Insert(disabledPlugin.Name)
@@ -170,7 +168,7 @@ func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2
 				klog.InfoS("Defaut plugin is explicitly re-configured and is overriding.", "plugin", defaultEnabledPlugin.Name)
 				// update the default plugin in place to preserve order
 				defaultEnabledPlugin = customPlugin
-				replacedCustomPlugins.Insert(customPlugin.Name)
+				delete(enabledCustomPlugins, defaultEnabledPlugin.Name)
 			}
 			enabledPlugins = append(enabledPlugins, defaultEnabledPlugin)
 		}
@@ -178,7 +176,7 @@ func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2
 
 	// append all the custom plugins which are not defined as default plugins.
 	for _, plugin := range customPluginSet.Enabled {
-		if !replacedCustomPlugins.Has(plugin.Name) {
+		if _, ok := enabledCustomPlugins[plugin.Name]; ok {
 			enabledPlugins = append(enabledPlugins, plugin)
 		}
 	}
```

Reply:
thumb up,  this looks better!

Reply:
mark this as resolved, and amend the commit directly.